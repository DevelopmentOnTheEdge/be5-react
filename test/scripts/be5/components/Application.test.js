import React from 'react';
import renderer from 'react-test-renderer';
import {shallow, mount, render} from 'enzyme';
import bus            from '../../../../src/scripts/be5/core/bus';
import App            from '../../../../src/scripts/be5/components/Application';
import testUtils      from '../testUtils'

test('snapshot', () => {
  testUtils.mockBe5();

  const component = renderer.create(
    <App />
  );
  expect(component.toJSON()).toMatchSnapshot();

  bus.fire("LoggedIn");
  // changeDocument('MainDocument', { component: StaticPage, value: "test" });
  // expect(component.toJSON()).toMatchSnapshot();

});

// test('test mount', () => {
//     mount(<App />);
// });
//
// test('test app', () => {
//   mockBe5();
//
//   const renderer = new ReactShallowRenderer();
//   renderer.render(<App />);
//
//   const result = renderer.getRenderOutput();
//   // expect(result.props.children.children).toEqual([
//   //     <SideBar ref="sideBar"/>,
//   //     <Document ref="document"/>
//   // ]);
//
//   const alertOptions = {
//     offset: 14,
//     position: 'top right',
//     theme: 'light',
//     time: 5000,
//     transition: 'fade',
//     icon: null
//   };
//   expect(result.props.children).toEqual([
//       <AlertContainer ref={a => this.msg = a} {...alertOptions } />,
//       <SplitPane split="vertical" defaultSize={280} >
//         <SideBar ref="sideBar"/>
//         <Document ref="document"/>
//       </SplitPane>
//   ]);
// });
//

