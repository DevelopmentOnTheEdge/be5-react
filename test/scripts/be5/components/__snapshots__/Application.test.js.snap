// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div>
  <div>
    <div
      className="s-alert-wrapper"
    />
  </div>
  <div
    className="SplitPane main-split-pane vertical "
    style={
      Object {
        "MozUserSelect": "text",
        "WebkitUserSelect": "text",
        "display": "flex",
        "flex": 1,
        "flexDirection": "row",
        "height": "100%",
        "left": 0,
        "msUserSelect": "text",
        "outline": "none",
        "overflow": "hidden",
        "position": "absolute",
        "right": 0,
        "userSelect": "text",
      }
    }
  >
    <div
      className="Pane vertical Pane1  "
      style={
        Object {
          "flex": "none",
          "outline": "none",
          "position": "relative",
          "width": 280,
        }
      }
    >
      <div
        className="side"
      >
        <div
          className="menuContainer"
        >
          <input
            className="searchField form-control"
            onChange={[Function]}
            placeholder="Filter..."
            type="text"
            value=""
          />
          <div
            className="menu"
          >
            <div
              className="menuNode"
            >
              <span
                className="rootMenuItem menuItemWithoutRef"
              >
                Loading...
              </span>
              <div />
            </div>
          </div>
        </div>
        <div
          className="menuFooter"
        >
          <img
            src={Object {}}
          />
        </div>
      </div>
    </div>
    <span
      className="Resizer vertical "
      onClick={[Function]}
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={Object {}}
    />
    <div
      className="Pane vertical Pane2  "
      style={
        Object {
          "flex": 1,
          "outline": "none",
          "position": "relative",
        }
      }
    >
      <div
        className="document-content"
        id="document-content___MainDocument"
      />
    </div>
  </div>
</div>
`;
